model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -76,79 +80,110 @@ def pattern(self):
-                                           'begin_mask': int64_array([1]),
-                                           'end_mask': int64_array([1]),
-                                           'new_axis_mask': int64_array([0]),
-                                           'shrink_axis_mask': int64_array([0]),
-                                           'ellipsis_mask': int64_array([0]),
-                                           }).create_node()
+        strided_slice_node = create_op_with_const_inputs(graph,
+                                                         StridedSlice,
+                                                         {
+                                                             1: int64_array([axis]),
+                                                             2: int64_array([axis + 1])
+                                                         },
+                                                         {
+                                                             'name': unsqueeze_name + '/StridedSlice',
+                                                             'begin_mask': int64_array([1]),
+                                                             'end_mask': int64_array([1]),
+                                                             'new_axis_mask': int64_array([0]),
+                                                             'shrink_axis_mask': int64_array([0]),
+                                                             'ellipsis_mask': int64_array([0]),
+                                                         })
         shape_node.out_port(0).connect(strided_slice_node.in_port(0))
-        slice_begin.out_port(0).connect(strided_slice_node.in_port(1))
-        slice_end.out_port(0).connect(strided_slice_node.in_port(2))
-
         cast_shape_to_float = Cast(graph, {'dst_type': np.float32}).create_node()
-
         strided_slice_node.out_port(0).connect(cast_shape_to_float.in_port(0))
-        cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
 
-        interp_node = Interpolate(graph,
-                                  dict(mode='nearest',
-                                       antialias=0, pads_begin=int64_array([0]),
-                                       pads_end=int64_array([0]), coordinate_transformation_mode='half_pixel',
-                                       nearest_mode='round_prefer_floor', cube_coeff=-0.75,
-                                       version='opset4', shape_calculation_mode='scales',
-                                       in_ports_count=4,
-                                       maybe_part_of_sequence=True)).create_node()
+        second_input_of_tile = match['tile'].in_port(1).get_connection().get_source().node
+        scale = float32_array([second_input_of_tile.value[d_idx]])
+        mul_node = create_op_with_const_inputs(graph, Mul, {1: scale}, {'name': unsqueeze_name + '/Mul'})
 
+        cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
```

Comment:
There is no need to cast to float because both operands for multiplication are integers.

Reply:
Fixed.